<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>917</width>
    <height>566</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_12">
   <item row="0" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="3">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>单价</string>
             </property>
            </widget>
           </item>
           <item row="0" column="5">
            <widget class="QLineEdit" name="editPrice"/>
           </item>
           <item row="1" column="3">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>总价</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="btnCalculate">
             <property name="text">
              <string>计算</string>
             </property>
            </widget>
           </item>
           <item row="1" column="5">
            <widget class="QLineEdit" name="editTotal"/>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>数量</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="editNum"/>
           </item>
           <item row="1" column="2">
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>13</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>28</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>十进制</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="editDecimal"/>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="btnDecimal">
             <property name="text">
              <string>转换为其他进制</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>二进制</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="editBinary"/>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="btnBinary">
             <property name="text">
              <string>转换为其他进制</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>十六进制</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLineEdit" name="editHex"/>
           </item>
           <item row="2" column="2">
            <widget class="QPushButton" name="btnHex">
             <property name="text">
              <string>转换为其他进制</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>28</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="0">
            <widget class="QLabel" name="label_7">
             <property name="minimumSize">
              <size>
               <width>10</width>
               <height>20</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>50</width>
               <height>100</height>
              </size>
             </property>
             <property name="text">
              <string>str1</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1" rowspan="2">
            <widget class="QComboBox" name="comboBox">
             <item>
              <property name="text">
               <string>G:\Qt5Book\QT5.9Study</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>G:\Qt5Book\QT5.9Study\qw.cpp</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>学生姓名，男，1984-3-4，汉族，山东</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="1" column="0" rowspan="2">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>str2</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="comboBox_2">
             <item>
              <property name="text">
               <string>\</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>.cpp</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>:</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>，</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>13</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QGroupBox" name="groupBox">
             <property name="title">
              <string>字符串</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_5">
              <item row="0" column="0">
               <widget class="QPushButton" name="btnAppend">
                <property name="text">
                 <string>Append</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QPushButton" name="btnPrepend">
                <property name="text">
                 <string>Prepend</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QPushButton" name="btnToUpper">
                <property name="text">
                 <string>toUpper</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QPushButton" name="btnToLower">
                <property name="text">
                 <string>toLower</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QPushButton" name="btnLeft">
                <property name="text">
                 <string>left</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QPushButton" name="btnRight">
                <property name="text">
                 <string>right</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QPushButton" name="btnSection">
                <property name="text">
                 <string>section</string>
                </property>
               </widget>
              </item>
              <item row="4" column="0">
               <widget class="QPushButton" name="btnSimplified">
                <property name="text">
                 <string>simplified</string>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <widget class="QPushButton" name="btnTrimm">
                <property name="text">
                 <string>trimmed</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_2">
             <property name="title">
              <string>数字</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_6">
              <item row="0" column="0">
               <widget class="QPushButton" name="btnCount">
                <property name="text">
                 <string>count</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QPushButton" name="btnSize">
                <property name="text">
                 <string>size</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QPushButton" name="btnIndexOf">
                <property name="text">
                 <string>indexOf</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QPushButton" name="btnLastIndexOf">
                <property name="text">
                 <string>lastIndexOf</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_3">
             <property name="title">
              <string>逻辑判断</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_7">
              <item row="0" column="0">
               <widget class="QPushButton" name="btnEndsWith">
                <property name="text">
                 <string>endsWith</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QPushButton" name="btnStartsWith">
                <property name="text">
                 <string>startsWith</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QPushButton" name="btnContains">
                <property name="text">
                 <string>contains</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QPushButton" name="btnIsNull">
                <property name="text">
                 <string>isNull</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QPushButton" name="btnIsEmpty">
                <property name="text">
                 <string>isEmpty</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_4">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>17</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="0" column="1" colspan="3">
            <widget class="QLineEdit" name="editResult"/>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QCheckBox" name="checkBox">
             <property name="text">
              <string>CheckBox</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>LabSpin</string>
             </property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QSpinBox" name="spinBox"/>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_9">
             <property name="minimumSize">
              <size>
               <width>10</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>50</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>结果</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <layout class="QGridLayout" name="gridLayout_8">
           <item row="0" column="0">
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>数量</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="spinBox_2">
             <property name="suffix">
              <string>kg</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>单价</string>
             </property>
            </widget>
           </item>
           <item row="0" column="4">
            <widget class="QDoubleSpinBox" name="doubleSpinBox">
             <property name="prefix">
              <string>$</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="btnCalculate_2">
             <property name="text">
              <string>计算</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="3">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>总价</string>
             </property>
            </widget>
           </item>
           <item row="1" column="4">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_2">
             <property name="prefix">
              <string>$</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_5">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>28</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout_9">
           <item row="1" column="0">
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>二进制</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QPushButton" name="btnHex_2">
             <property name="text">
              <string>转换为其他进制</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="btnDecimal_2">
             <property name="text">
              <string>转换为其他进制</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="spinBox_4">
             <property name="prefix">
              <string>Bin </string>
             </property>
             <property name="displayIntegerBase">
              <number>2</number>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_16">
             <property name="text">
              <string>十六进制</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="spinBox_5">
             <property name="prefix">
              <string>Hex </string>
             </property>
             <property name="displayIntegerBase">
              <number>16</number>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_15">
             <property name="text">
              <string>十进制</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="btnBinary_2">
             <property name="text">
              <string>转换为其他进制</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="spinBox_3">
             <property name="prefix">
              <string>Dec </string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>Slider and Color</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_13">
          <item row="1" column="0">
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>Blue</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSlider" name="sliderAlpha">
            <property name="maximum">
             <number>155</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="2" rowspan="4">
           <widget class="QTextEdit" name="textColor"/>
          </item>
          <item row="2" column="1">
           <widget class="QSlider" name="sliderGreen">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="6" column="0" colspan="3">
           <widget class="QSlider" name="sliderRed_2">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksAbove</enum>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_19">
            <property name="text">
             <string>Green</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSlider" name="sliderRed">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>Alpha</string>
            </property>
           </widget>
          </item>
          <item row="7" column="0" colspan="3">
           <widget class="QScrollBar" name="horizontalScrollBar">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>Red</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSlider" name="sliderBlue">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>200</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="progressBar">
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>24</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>Dial and LCD num</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_11">
          <item row="0" column="0">
           <widget class="QDial" name="dial">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="sliderPosition">
             <number>0</number>
            </property>
            <property name="tracking">
             <bool>true</bool>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="invertedAppearance">
             <bool>false</bool>
            </property>
            <property name="invertedControls">
             <bool>false</bool>
            </property>
            <property name="wrapping">
             <bool>false</bool>
            </property>
            <property name="notchTarget">
             <double>10.000000000000000</double>
            </property>
            <property name="notchesVisible">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLCDNumber" name="lcdNumber">
            <property name="value" stdset="0">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QGroupBox" name="groupBox_6">
            <property name="title">
             <string>LCD 显示进制</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_10">
             <item row="0" column="0">
              <widget class="QRadioButton" name="radioButton">
               <property name="text">
                <string>十进制</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QRadioButton" name="radioButton_2">
               <property name="text">
                <string>二进制</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QRadioButton" name="radioButton_4">
               <property name="text">
                <string>八进制</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QRadioButton" name="radioButton_3">
               <property name="text">
                <string>十六进制</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
